115:Parameter error: expression must be absolute (parameter 3)
138:Invalid mnemonic 'plt'

--- libs/context/src/asm/make_ppc32_sysv_macho_gas.S.orig
+++ libs/context/src/asm/make_ppc32_sysv_macho_gas.S
@@ -111,7 +111,7 @@
     # load LR into R4
     mflr  r4
     # compute abs address of label finish
-    addi  r4, r4, finish - 1b
+    addi  r4, r4, lo16(finish - 1b)
     # restore LR
     mtlr  r0
     # save address of finish as return-address for context-function
@@ -134,4 +134,4 @@
     # exit code is zero
     li  r3, 0
     # exit application
-    bl  _exit@plt
+    bl  __exit
